cutil/logger/logger.go

```diff
@@ -137,6 +125,20 @@ func Warning(format string, a ...interface{}) {
 	}
 }
 
+func extractLoggerArgs(format string, a ...interface{}) ([]interface{}, io.Writer) {
```

Comment:
I totally get your reasoning and I think that we should do better in the very near future. The sole purpose for supporting an `io.Writer` option in the `a ...interface{}` parameter of each logger function was to enable testing of these logger functions and of the process annotations delivered in this PR (hence no documentation on this).

@kris-nova: the `logger.FInfo` example makes total sense, but if you look at the overall architecture of the `logger` package, you'll see that the implementations decide which writer to choose based on the state of package variables (unless you're in testing mode). IMO, if we declare `io.Writer` parameters in the function signatures, we'll have to rethink this behavior.

As it's submitted here, we don't break with the logger API and we didn't have to change the design. I'd vote for accepting this PR to include annotations for long-running tasks and then reconsider the logger design for the next milestone. Thoughts?